client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -12,33 +14,35 @@ interface Params< FlowSteps extends StepperStep[] > {
 }
 
 export const useStepNavigationWithTracking = ( {
-	flow,
+	flow: { useTracksEventProps, useStepNavigation, name: flowName, variantSlug },
 	currentStepRoute,
 	navigate,
 	steps,
 }: Params< ReturnType< Flow[ 'useSteps' ] > > ) => {
-	const stepNavigation = flow.useStepNavigation(
+	const stepNavigation = useStepNavigation(
 		currentStepRoute,
 		navigate,
 		steps.map( ( step ) => step.slug )
 	);
 	const intent =
 		useSelect( ( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getIntent(), [] ) ?? '';
+	const tracksEventPropsFromFlow = useTracksEventProps?.();
 
-	const shouldTrackSubmit = flow.use__Temporary__ShouldTrackEvent?.( 'submit' ) ?? false;
-
-	return {
-		...stepNavigation,
-		submit: ( providedDependencies: ProvidedDependencies = {}, ...params: string[] ) => {
-			shouldTrackSubmit &&
+	return useMemo(
```

Comment:
Just curious, why do we need `useMemo`?

Reply:
It will generate a new object on every call otherwise, and I cannot control how often this gets called and what side effects it may create. So "referential integrity"?